services/frontend/src/scenes/Profile/ProfileLayout.js

```diff
@@ -122,18 +124,85 @@ const PersonalDetails = ({profile}) => {
     );
 };
 
-const Skills = ({sectionName, profile}) => (
-    <>
-        <h2>{sectionName}</h2>
-        <Paper className="profile-page-card profile-card-skills">
-            <SkillBadges
-                displayCount={profile.skills.length}
-                skills={profile.skills}
-            />
-        </Paper>
-    </>
-);
+const Skills = ({sectionName, profile, skills}) => {
+    const [editSkillsDialogOpen, setEditSkillsDialogOpen] = React.useState(false);
+    const [profileUpdated, updateProfile] = useState(profile);
+
+    const openDialog = () => {
+        setEditSkillsDialogOpen(true);
+    };
+
+    const closeDialog = () => {
+        setEditSkillsDialogOpen(false);
+    };
+
+    const handleSubmit = (updatedSkills) => {
+        removeSkills(updatedSkills);
+        addSkills(updatedSkills);
+        closeDialog();
+    };
+
+    const handleCancel = () => {
```

Comment:
If you don't have plans to expand on this function in the future, I would just remove this and replace it's references with closeDialog. Doesn't really seem needed.

Reply:
I goes into a loop if I directly have it, but I think it would be useful when I do the notification.

Reply:
Hmm, doesn't loop for me but if you are going to expand this to be different from `closeDialog()` it can be left.